desktop_version/cmake/GenerateScriptCode.cmake

```diff
@@ -0,0 +1,40 @@
+SET(SCRIPTS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src/scripts")
+SET(OUT_DIR "${SCRIPTS_DIR}/generated")
+
+# Generate code for script files so they can be #included
+FILE(GLOB SCRIPTS "${SCRIPTS_DIR}/*.vsc")
+FOREACH(SCRIPT_FILE ${SCRIPTS})
+	GET_FILENAME_COMPONENT(SCRIPT_NAME "${SCRIPT_FILE}" NAME_WE)
+
+	GET_FILENAME_COMPONENT(OUT_FILE "${SCRIPT_FILE}" NAME)
+	SET(OUT_FILE "${OUT_DIR}/${OUT_FILE}.cpp")
+
+	# Only regenerate what changed
+	IF(EXISTS "${OUT_FILE}" AND "${OUT_FILE}" IS_NEWER_THAN "${SCRIPT_FILE}")
+		CONTINUE()
+	ENDIF()
+
+	FILE(READ ${SCRIPT_FILE} SCRIPT_CONTENT)
+
+	# Wrap script in initialization code
+	STRING(REGEX REPLACE "\r?\n" ";" SCRIPT_LINES "${SCRIPT_CONTENT}")
+	SET(SCRIPT_CONTENT "if (SDL_strcmp(t, \"${SCRIPT_NAME}\") == 0) {\n\tstatic const char* lines[] = {\n")
+	FOREACH(LINE ${SCRIPT_LINES})
+		# Skip comment lines
+		IF (LINE MATCHES "^#")
+			CONTINUE()
+		ENDIF()
+
+		# Remove inline comments
+		STRING(REGEX REPLACE "#.+$" "" LINE "${LINE}")
```

Comment:
It's nice that you specifically added code to deal with comments (and want to preserve them), but the scripting engine simply ignores all invalid commands, including comments. So I'm not really sure how necessary this is.

Reply:
That's good to know. One additional benefit of this is that the size of the static string arrays will be (marginally) smaller. In any case, leaving the comments out of the generated code is cleaner to me but I could go either way.

Reply:
Does the scripting engine ignore invalid text at the end of an otherwise valid line?

Reply:
Yes, as long as you always close the arguments to the command with a parenthesis at the end.